### PR TITLE
Much faster uninstall of applications

### PR DESCRIPTION
**Pull request by PFCKrutonium**
Changed the wildcard conventions, and changed how TRON runs the command
- This should drastically speed up the removal of programs by name.

Source is available if you want to see it, but it should be kept private.

### CODE REVIEW

**Comment by nemchik** [comment]
Please make sure to keep the dry run code in there, and also does this method offer any logging output? I don't see anything obvious in your code indicating that the changes made will be logged for review later.

**Comment by PFCKrutonium** [comment]
When the application runs, it will drop a log.txt file in the CD containing the time/date/application that is being removed.

**Comment by nemchik** [comment]
I'm not sure if this part of the code uses %LOGPATH% as the current directory, but does your tool offer to output to a  specific location? Or is the current directory log.txt the only option? If so can you add an extra line to move the log.txt into the %LOGPATH% and name it something that wouldn't be used by another tool so that we know what it is?

**Comment by PFCKrutonium** [comment]
Sure, I'll be back shortly.

**Comment by nemchik** [comment]
Awesome, is this executable something you wrote?

**Comment by PFCKrutonium** [comment]
Yes - Specifically because running all these WMIC commands take forever, and I knew I could do it in a fraction of the time :+1:

**Comment by PFCKrutonium** [comment]
I've made the change, but I can't add to this, due to a merge confict

And I am not very good with Git,

**Comment by nemchik** [comment]
You can lookup how to rebase and that should let you delete this PR and make a new one with your changes.

I'm not the author BTW, just trying to be helpful :)

**Comment by PFCKrutonium** [comment]
Fair enough, will look

**Comment by vocatus** [comment]
Hi guys,

Been backlogged with other work lately, but I'm still monitoring Git and the subreddit. Can you tl;dr this change for me? Additionally, I'm not entirely comfortable with adding a closed-source tool to the main workflow; are you opposed to open-sourcing it? (I did notice the mention that you could share the code with me for review).

Thanks for your work and contributions, people like you two are the reason Tron keeps getting better.

**Comment by nemchik** [comment]
tl;dr @PFCKrutonium made an exe that handles uninstalls faster than WMIC and is willing to share the source code but prefers it not be public.

The changes look good to me, i as just interjecting about keeping the dry run ability and logging.

**Comment by vocatus** [comment]
Is there any way we can get it to run *without* dependency on .NET? A significant number of systems don't have it installed (although I'm thinking of Win7 and older; correct me if later versions of Windows come with it pre-bundled).

**Comment by nemchik** [comment]
I believe Win8 and newer come with .NET 4.5.x preinstalled, but they do not include 3.5 which must be installed separately. I think 7 and below do not come with .NET preinstalled.

**Comment by PFCKrutonium** [comment]
Windows 7 had 3.5 and older installed, and .Net 4 is installed via an update, Windows 8 and higher has .net 4 and newer installed, and prompts to install .net 3.5 and older when you run a program that actually uses it.

**Comment by vocatus** [comment]
I really like this idea, but can't include it if it depends on .NET unfortunately. With Tron I try to target the lowest common denominator, and when there's a choice between speed + dependency, or going slower but more reliable, I'll go with the more reliable option.

**Comment by PFCKrutonium** [comment]
It wouldn't be difficult to find out if .net is installed and do one or the other, 


reg query "HKLM\SOFTWARE\Microsoft\NET Framework Setup\NDP\v3.5"| findstr Install

**Comment by nemchik** [comment]
That's not a bad idea, but then you have to maintain two sets of code in tron; one to work when there is no .net and the other to work when there is. This idea becomes complicated when looking at your changes to resources/stage_2_de-bloat/programs_to_target_by_name.txt because the switch from `%%` to `*`. This might be easier if your program were able to accept the same wildcard that wmic uses, just so that switching back and forth would be easier. Would it be possible for you to update your program for that purpose?

**Comment by PFCKrutonium** [comment]
I can, as long as no programs we are removing use * in the name. Is this the case?

**Comment by vocatus** [comment]
This is great, thanks PFCKrutonium. 

Let me test it out (might not be till this weekend) and assuming it works I'll add it in.

File: tron.bat [hunk 1]
```diff
@@ -1052,7 +1052,8 @@ title TRON v%SCRIPT_VERSION% [stage_2_de-bloat] [Remove bloatware by name]
 call :log "%CUR_DATE% %TIME%    Attempt junkware removal: Phase 2 (wildcard by name)..."
 call :log "%CUR_DATE% %TIME%    Customize here: \resources\stage_2_de-bloat\oem\programs_to_target_by_name.txt"
 :: Search through the list of programs in "programs_to_target.txt" file and uninstall them one-by-one
-if /i %DRY_RUN%==no FOR /F "tokens=*" %%i in (stage_2_de-bloat\oem\programs_to_target_by_name.txt) DO echo   %%i && echo   %%i...>> "%LOGPATH%\%LOGFILE%" && %WMIC% product where "name like '%%i'" uninstall /nointeractive>> "%LOGPATH%\%LOGFILE%"
+::if /i %DRY_RUN%==no FOR /F "tokens=*" %%i in (stage_2_de-bloat\oem\programs_to_target_by_name.txt) DO echo   %%i && echo   %%i...>> "%LOGPATH%\%LOGFILE%" && %WMIC% product where "name like '%%i'" uninstall /nointeractive>> "%LOGPATH%\%LOGFILE%"
+call resources\stage_2_de-bloat\Remove_by_name.exe /I=\resources\stage_2_de-bloat\programs_to_target_by_name.txt
```
Comment (vocatus) [inline tron.bat]: Can you update this line place remove_by_name.exe in the oem subfolder of stage_2_de-bloat?

File: resources/stage_2_de-bloat/programs_to_target_by_name.txt [hunk 2]
```diff
@@ -1,222 +1,222 @@
...
-%%TidyNetwork%%
-%%Toolbar%%
-%%Trial%%
-%%Virtumundo%%
-%%VirusProtectPro%%
-%%WeatherBug%%
-%%WhenUsave%%
-%%Zango%%
-180 Solution%%
+*180Search*
```
Comment (vocatus) [inline resources/stage_2_de-bloat/programs_to_target_by_name.txt]: Can you update this file to revert to the old-style wildcards? (unless your program accepts both, then we can go with astericks since it's more intuitive)

**Comment by vocatus** [comment]
Hi PFCKrutonium, I haven't forgotten about this and would still like to do it. Can you re-base it to match the current code and remove anything that alters the de-bloat lists? (I'll do that manually).

**Comment by PFCKrutonium** [comment]
I am not at all sure how to do this, so I am going to make a new PR.

**Comment by nemchik** [comment]
Rebasing is sometimes really easy and other times it's extremely confusing. Takes a couple tries to master.

Looking forward to seeing your new PR though :)

**Comment by PFCKrutonium** [comment]
I should be able to make it probably tomorrow - See you then!
